src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -533,12 +520,9 @@ public bool HasPendingSymbolAnalysis(AnalysisScope analysisScope, CancellationTo
 
         private ImmutableArray<SymbolDeclaredCompilationEvent> GetPendingSymbolDeclaredEvents(SyntaxTree tree, CancellationToken cancellationToken)
         {
-            Debug.Assert(tree != null);
```

Comment:
Why remove the assertion?

Reply:
Do we need the assertion for nullable enabled files where the type guarantees non-null? I only had the assert because the file was not nullable enabled. I am not sure what is our preference here.. I can add it back if you feel it is helpful.

Reply:
Nullable annotations offer a weaker guarantee than assertions. If all callers are in nullable-enabled files, then it may be fine to remove the assertion. Otherwise, I'd leave it.

Reply:
Thanks. This is a private method only used in this file, so doesn't need an assertion.